dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/programrule/engine/ProgramNotificationInstanceServiceTest.java

```diff
@@ -136,6 +169,36 @@ void testGetProgramNotificationInstance() {
     assertFalse(instances.isEmpty());
   }
 
+  @Test
+  void testGetProgramNotificationInstanceScheduledForToday() throws ParseException {
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+    String formattedDate = sdf.format(new Date());
+
+    Date todayDate = sdf.parse(formattedDate);
+    ProgramNotificationInstanceParam param =
+        ProgramNotificationInstanceParam.builder().scheduledAt(todayDate).build();
+
+    List<ProgramNotificationInstance> instancesForToday =
+        programNotificationInstanceService.getProgramNotificationInstances(param);
+
+    assertFalse(instancesForToday.isEmpty());
+
+    // Only one instance is scheduled for today's date
+    assertEquals(1, instancesForToday.size());
+    ProgramNotificationInstance instanceForToday = instancesForToday.get(0);
+    assertEquals(instanceForToday.getScheduledAt(), todayDate);
+  }
+
+  @Test
+  @Timeout(value = 20, unit = TimeUnit.SECONDS)
+  void testShouldGetAndSendScheduledNotificationInstanceWithoutTimeout() {
+    List<ProgramNotificationInstance> instances = programNotificationInstanceService.getAll();
```

Comment:
I don't fully get this test.
We do not need to have `getAll` implemented, to assert the same you could check for today and tomorrow notifications and see if they are there.

On the timeout, it is really a shot in the dark, this test with this timeout was probably passing even before your changes, so I am not sure it makes sense to have it at all.